stock_cancel/models/stock_quant.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (c) 2012 Andrea Cometa All Rights Reserved.
+#                       www.andreacometa.it
+#                       openerp@andreacometa.it
+#    Copyright (C) 2013 Agile Business Group sagl (<http://www.agilebg.com>)
+#    Ported to new API by Alexis de Lattre <alexis.delattre@akretion.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published
+#    by the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from odoo import api, models
+
+
+class StockQuant(models.Model):
+    _inherit = 'stock.quant'
+
+    @api.multi
+    def _revert(self):
+        for quant in self:
+            previous_move = quant.history_ids[-1]
```

Comment:
@aheficent Are you 100% sure the move -1 in history would be the good one?

Reply:
@rousseldenis No, I'm not. Functional test from my side worked for simple pickings (deliveries, shipments or internal moves when only 2 locations are involved). Pending to check chained stock moves thought. I'll add some tests trying to cover most common cases.